File: OOPS in Java/ConstructorChaining.java
class Employee {
    private int id;
    private String name;
    private String department;
    private String city;
    private double salary;

    // Constructor chaining
    // Constructor with all parameters
    public Employee(int id, String name, String department, String city, double salary) {
        this.id = id;
        this.name = name;
        this.department = department;
        this.city = city;
        this.salary = salary;
    }

    // Constructor with ID and Name
    public Employee(int id, String name) {
        this(id, name, "Unknown", "Unknown", 0.0);
    }

    // Constructor with ID, Name, and Department
    public Employee(int id, String name, String department) {
        this(id, name, department, "Unknown", 0.0);
    }

    // Constructor with ID, Name, Department, and City
    public Employee(int id, String name, String department, String city) {
        this(id, name, department, city, 0.0);
    }

    // Default constructor
    public Employee() {
        this(0, "Unknown", "Unknown", "Unknown", 0.0);
    }

    // Method to display employee details
    public void display() {
        System.out.println("ID: " + id);
        System.out.println("Name: " + name);
        System.out.println("Department: " + department);
        System.out.println("City: " + city);
        System.out.println("Salary: " + salary);
    }

    public static void main(String[] args) {
      
        Employee emp1 = new Employee(1, "John Doe", "Engineering", "New York", 75000);
        Employee emp2 = new Employee(2, "Jane Smith", "Marketing");
        Employee emp3 = new Employee(3, "Alice Johnson");
        Employee emp4 = new Employee();
      
        emp1.display();
        System.out.println();
        emp2.display();
        System.out.println();
        emp3.display();
        System.out.println();
        emp4.display();
    }
}

File: OOPs in Cpp/ConstructorChaining.cpp
#include <iostream>
#include <string>
using namespace std;

class Employee {
private:
    int id;
    string name;
    string department;
    string city;
    double salary;
public:

    Employee(int id , string name, string department, string city, double salary) {
        this->id = id;
        this->name = name;
        this->department = department;
        this->city = city;
        this->salary = salary;
    }

    Employee(int id, std::string name) : 
        Employee(id, name, "Unknown", "Unknown", 0.0) 
            {}

    Employee(int id, std::string name, std::string department) :
        Employee(id, name, department, "Unknown", 0.0) 
            {}

    Employee(int id, std::string name, std::string department, std::string city) : 
        Employee(id, name, department, city, 0.0) 
            {}

    Employee() : 
        Employee(0, "Unknown", "Unknown", "Unknown", 0.0)
            {}

    void display() {
        cout << "ID: " << id << endl;
        cout << "Name: " << name << endl;
        cout << "Department: " << department << endl;
        cout << "City: " << city << endl;
        cout << "Salary: " << salary << endl;
    }
};

int main() {
    // Using different constructors
    Employee emp1(1, "John Doe", "Engineering", "New York", 75000);
    Employee emp2(2, "Jane Smith", "Marketing");
    Employee emp3(3, "Alice Johnson");
    Employee emp4;

    emp1.display();
    cout << endl;
    emp2.display();
    cout << endl;
    emp3.display();
    cout << endl;
    emp4.display();

}
